fs/graph.go

```diff
@@ -269,6 +269,28 @@ func (g *Graph) AddReferencingTests() []*File {
 	return tests
 }
 
+// AddEphemeralUpstreams brings any upstream models that are ephemeral into the graph
+func (g *Graph) AddEphemeralUpstreams() {
+	for _, node := range g.nodes {
+		if node.file.Type != ModelFile {
+			continue
+		}
+
+		for upstream := range node.file.upstreams {
```

Comment:
Can we extract this to a private method on the graph maybe called `addEphemeralUpstreamsFor(file *File)` then call this where this loop is.

That way after we add the upstreamNode on line 289 in the loop we can call it rescurively with `g.addEphemeralUpstreamsFor(upstreamNode)` - that way if an ephemeral model uses another ephemeral model we capture it too